data/tutorials/language/0it_06_imperative.md

```diff
@@ -331,6 +331,140 @@ In this example, the first two expressions are `print_endline` function calls, w
 
 Here, the semicolon after 42 is ignored.
 
+**`begin … end` expressions**
+
+Imagine we want to write a function that:
+1. Has an `int` reference parameter containing value _n_
+2. Updates the reference's contents to _2 &times; (n + 1)_
+
+This is arguably convoluted and does not work:
+```ocaml
+let f r = r := incr r; 2 * !r;;
+Error: This expression has type unit but an expression was expected of type int
+```
+
+But here is how it can be made to work:
+```ocaml
+# let f r = r := begin incr r; 2 * !r end;;
+val f : int ref -> unit = <fun>
+```
+
+The error came from assign `:=` associating more that semicolum `;`. Here is what we want to do, in order:
+1. Increment `r`
+2. Compute `2 * !r`
+3. Assign into `r`
+
+Remember the value of a semicolon-separated sequence is the value of its last expression. Grouping the first two steps fix the error. Usually, this is done using parentheses, but surrounding it with `begin … end` works, too. The latter is preferred to highlight expressions that have side effects.
+
+**Fun fact**: `begin … end` and parenthesis are badly the same:
+```ocaml
+# begin end;;
+- : unit = ()
+```
+
+### Conditional Execution
```

Comment:
This title is indirect, doesn't make clear that this is about If-Then-Else. From the title alone, it could also be about match expressions.